app/components/listing/RelativeOpeningTime.jsx

```diff
@@ -17,18 +24,74 @@ class RelativeOpeningTime extends React.Component {
     clearInterval(this.tick);
   }
 
+  static parseSchedule(schedule_days) {
+    if (!schedule_days) return { text: '', classes: '' };
+
+    const STATUS_CLOSED = 'status-red';
+    const STATUS_OPEN = 'status-green';
+    const STATUS_CAUTION = 'status-amber';
+
+    const currentDate = new Date();
+    const currentDay = currentDate.getDay();
+    const currentDayString = RelativeOpeningTime.daysOfWeek[currentDay];
+
+    const todayHours = schedule_days.filter(hours => hours.day === currentDayString);
+    const tomorrowHours = schedule_days.filter(hours => hours.day === RelativeOpeningTime.daysOfWeek[currentDay + 1 > 6 ? 0 : currentDay + 1]);
+    const daysOpen24Hours = schedule_days.filter(hours => hours.opens_at === 0 && hours.closes_at === 2359)
```

Comment:
A refactoring suggestion: We could pull out a function `is24Hours()` that takes in an `hours` object and returns true iff it's open 24 hours. This same check appears several times in this function.

Reply:
Huh? This is the only time this is checked, after that we  check if there are 7 days or if today or tomorrow is 24h